### DISCUSSION TITLE
Create a runtime instance every time or share one?

### DISCUSSION THREAD

**Discussion by ScottLinnn** [tokio-rs/tokio#4799] [Q&A] [answered]
I want to let all my microservices run on the current thread and I don't want to use the main macro. Should they use the same single-threaded runtime instance or share the same one?

**Comment by Noah-Kennedy** [comment, marked as answer]
If you want these microservices to run on the same thread, you should put them all on once shared `current_thread` runtime. If you want them to each have their own thread, you should use a separate `current_thread` runtime for each.
